packages/color-scale-block/src/ColorScaleBlock.tsx

```diff
@@ -0,0 +1,539 @@
+                            ...blockSettings,
+                            'color-input': colorsWithNewWidths,
+                        });
+                        setDisplayableItems(colorsWithNewWidths);
+                    } else {
+                        return setDisplayableItems(blockSettings['color-input']);
+                    }
+                } else {
+                    populatePlaceholderSquares();
+                }
+            } catch (error) {
+                console.error(error);
+            }
+        }
+    }, [blockSettings]);
+
+    const handleResizeStop = () => {
+        draggingId.current = null;
+    };
+    const handleResizeStart = (event: MouseEvent, index?: number, currentColor?: ColorProps) => {
+        if (dragStartPos) {
+            dragStartPos.current = event.clientX;
+            dragStartWidth.current = currentColor?.width;
+        }
+        draggingId.current = index;
+    };
+
+    const handleResize: MouseEventHandler = (event: MouseEvent) => {
+        clearTimeout(timerToUpdateBlockSettings.current);
+
+        if (draggingId.current !== null) {
+            const id = draggingId.current;
+            if (!lastDragPos.current) {
+                lastDragPos.current = event.clientX;
+            }
+
+            if (event.clientX < lastDragPos?.current) {
+                if (lastDragPos.current - event.clientX >= 8) {
+                    lastDragPos.current = event.clientX;
+
+                    let valuesChanged = false;
+
+                    if (!dragStartPos.current) {
+                        return;
+                    }
+
+                    const movementSinceStart = dragStartPos.current - event.clientX;
+
+                    const colorsBeforeCurrent = displayableItems
+                        ?.filter((diValue, diIndex) => {
+                            if (id !== undefined && diIndex < id) {
+                                return true;
+                            }
+                            return false;
+                        })
+                        .reverse();
+
+                    const newDisplayableItems = displayableItems?.map((diValue, diIndex) => {
+                        if (diValue && diIndex === id) {
+                            if (diValue.width && diValue.width >= 16) {
+                                // need to make sure it's 16 because we're going to decrease width
+                                // by 8 pixels, and the minimum width is 8 pixels
+                                if ((dragStartWidth.current ?? 0) - movementSinceStart >= 8) {
+                                    diValue.width = (dragStartWidth.current ?? 0) - movementSinceStart;
+
+                                    valuesChanged = true;
+                                }
+                            } else {
+                                let needToShrinkColor = true;
+                                colorsBeforeCurrent?.map((adjacentColor) => {
```

Comment:
why using map loop and why returning a value when map is not used. Maybe use foreach or every()